### PR TITLE
[BugFix][MRV2] Align Gumbel sampling with upstream logits cache

### PR DESCRIPTION
**Pull request by freyfwt**
### What this PR does / why we need it?

Upstream MRv2 draft sampling passes `logits_cache` and `logits_cache_col`, but the Ascend Gumbel override still exposes `output_processed_logits`, causing keyword-argument failures. Align the cache arguments and accept the newer `is_drafting` argument, defaulting to `False` for the currently verified vLLM revision.

Cache logits before temperature scaling and use both cache strides. Scale cached draft logits in verification and both categorical resampling stages to match upstream block statistics. Preserve the optimized resample implementation from main. Honor the upstream draft-noise position salt when `is_drafting=True`.

Explicitly bind DSpark's imported `gumbel_sample` to the Ascend implementation so probabilistic drafting uses the NPU operator even when DSpark is imported before the worker patches. DSpark's greedy path continues to use argmax.

### Does this PR introduce _any_ user-facing change?

Fixes MRv2 draft sampling with the upstream cache interface and non-unit temperatures, including DSpark's probabilistic path. No new configuration options. FP64 sampling remains unsupported on NPU.

### How was this patch tested?

**Dual-version integration validation is currently blocked on vLLM v0.28.0 by pre-existing repository startup incompatibilities.** The table below tests the same PR commit, `3e5ab18982a695fd5dcdc97117bba4b1599c7e5b`, with separate upstream checkouts and verified source paths.

| vLLM baseline | CPU regression | Real A3 NPU regression | Full mypy (Python 3.10 / 3.11 / 3.12) |
| --- | --- | --- | --- |
| `.github/vllm-main-verified.commit`: [`e6bfe03ad73a`](https://github.com/vllm-project/vllm/commit/e6bfe03ad73a3330cb427885aa90d97a12e1c704) | 6 passed | 68 passed: 41 Gumbel/DSpark + 27 resample | All passed |
| `v0.28.0` / `releases/v0.28.0`: [`2cf0a6915ce5`](https://github.com/vllm-project/vllm/commit/2cf0a6915ce544dc493a0990f2ea38d81601128a) | Collection blocked | Collection blocked | All passed |

Commands run separately with each upstream checkout on `PYTHONPATH`:

```bash
python -m pytest -q tests/ut/patch/worker/test_patch_v2_gumbel.py
python -m pytest -q tests/e2e/pull_request/one_card/test_gumbel_sampling.py tests/e2e/nightly/single_node/ops/singlecard_ops/triton/test_resample.py
for python_version in 3.10 3.11 3.12; do
    tools/mypy.sh 1 "$python_version"
done
```

The NPU environment used Ascend A3 and `triton-ascend==3.2.2`. Coverage includes raw caches, scalar/per-token columns, strided/BF16 caches, invalid requests, draft noise separation, DSpark dispatch, and temperature equivalence across single-block and cross-block resampling.

On v0.28.0, standard CPU and NPU collection fails before executing these tests: `vllm_ascend/patch/platform/patch_use_v2_model_runner.py` accesses the absent `VllmConfig._get_v1_model_runner_unsupported_features`. The same NPU collection failure was reproduced on the unchanged PR base `b0c49dc7a884d3d343f6a556380f11b74191c3e8`; this PR does not modify that patch. Separately loading the existing worker Triton patch also encounters the absent `vllm.v1.attention.ops.pcp`, imported by unchanged `vllm_ascend/attention/attention_v1.py`.

Supplemental **isolated kernel validation** on v0.28.0 passed **67 tests, with 1 deselected**, using the same two NPU files with `--noconftest -k 'not test_dspark_uses_ascend_gumbel'`. This bypasses repository-wide patch initialization and excludes the DSpark patch-binding integration case. It validates the kernels on 0.28, but **does not establish DSpark integration or a passing dual-version integration matrix**.

- `bash format.sh ci` passed on this unchanged PR commit.
- CPU UT routes to CPU; the Gumbel PR test routes to A2 single-card CI. Manual hardware validation used A3, not A2.
- These are focused compatibility checks, not the full community NPU suite. At verification time, upstream `.github/vllm-release-tag.commit` still named `v0.27.1`; the supplemental release check above explicitly uses the requested `v0.28.0`.

- vLLM main: https://github.com/vllm-project/vllm/commit/b2f685834a6456197e7033966fdef52a23f1abcd

### CODE REVIEW

**Review by weijinqian0 — DISMISSED** [review]

**Comment by freyfwt** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Review by weijinqian0 — APPROVED** [review]
